### PR TITLE
ddl: Revoke the session when the DDL will close (#3454)

### PR DESCRIPTION
**Pull request by zimulala**
* ddl: revoke the session and get env

Merge #3454 to master.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: ddl/owner_manager.go [hunk 1]
```diff
@@ -112,6 +114,20 @@ func (m *ownerManager) SetBgOwner(isOwner bool) {
...
+func setManagerSessionTTL() error {
+	ttlStr := os.Getenv("manager_ttl")
+	if len(ttlStr) == 0 {
+		return nil
+	}
+	ttl, err := strconv.Atoi(ttlStr)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	ManagerSessionTTL = ttl
```
Comment (tiancaiamao) [inline ddl/owner_manager.go]: In my opinion, use environment to control program's behavior is really a bad idea.
Why is this needed? is there any better way?

File: ddl/owner_manager.go [hunk 2]
```diff
@@ -162,8 +178,13 @@ func (m *ownerManager) campaignLoop(ctx goctx.Context, etcdSession *concurrency.
 				return
 			}
 		case <-ctx.Done():
-			log.Infof("[ddl] break %s campaign loop", key)
-			return
+			// Revoke the session lease.
+			// If revoke takes longer than the ttl, lease is expired anyway.
+			ctx, cancel := goctx.WithTimeout(goctx.Background(),
+				time.Duration(ManagerSessionTTL)*time.Second)
+			_, err = m.etcdCli.Revoke(ctx, etcdSession.Lease())
```
Comment (tiancaiamao) [inline ddl/owner_manager.go]: Why call with a timeout context, will Revoke would hang forever?

File: ddl/owner_manager.go (hunk 1 above)
Comment (zimulala) [inline ddl/owner_manager.go]: It's used for testing.

Comment (tiancaiamao) [inline ddl/owner_manager.go]: But the real code is affected, if user specify a "manager_ttl=0" in the environment, 
and get strange behavior, we have no way to know!

File: ddl/owner_manager.go (hunk 2 above)
Comment (zimulala) [inline ddl/owner_manager.go]: It revokes util its result is successful or failed, and etcd is so used.

**Comment by tiancaiamao** [comment]
LGTM
